starlette/applications.py

```diff
@@ -137,19 +130,17 @@ def host(
     ) -> None:  # pragma: no cover
         self.router.host(host, app=app, name=name)
 
-    def add_middleware(
-        self, middleware_class: type, **options: typing.Any
-    ) -> None:  # pragma: no cover
+    def add_middleware(self, middleware_class: type, **options: typing.Any) -> None:
+        if self.middleware_stack is not None:  # pragma: no cover
```

Comment:
Can't we add a test for this case? :thinking:

Reply:
Seems like testing for the sake of testing but if you want I can add it.